spectractor/extractor/spectrum.py

```diff
@@ -503,6 +511,116 @@ def plot_spectrogram(self, ax=None, scale="lin", title="", units="Image units",
         if parameters.PdfPages:  # pragma: no cover
             parameters.PdfPages.savefig()
 
+    def plot_spectrum_summary(self, xlim=None, figsize=(12, 12), saveAs=''):
+        """Plot spectrum with emission and absorption lines.
+
+        Parameters
+        ----------
+        ax: Axes, optional
+            Axes instance (default: None).
+        label: str
+            Label for the legend (default: '').
+        xlim: list, optional
+            List of minimum and maximum abscisses (default: None)
+        live_fit: bool, optional
+            If True the spectrum is plotted in live during the fitting procedures
+            (default: False).
+        force_lines: bool
+            Force the over plot of vertical lines for atomic lines if set to True (default: False).
+        saveAs : str, optional
+            Path to save the figure to, if specified.
+
+        Examples
+        --------
+        >>> s = Spectrum(file_name='tests/data/reduc_20170530_134_spectrum.fits')
+        >>> s.plot_spectrum_summary()
+        """
+        if not np.any([line.fitted for line in self.lines.lines]):
+            fwhm_func = interp1d(self.chromatic_psf.table['lambdas'],
+                                 self.chromatic_psf.table['fwhm'],
+                                 fill_value=(parameters.CALIB_PEAK_WIDTH, parameters.CALIB_PEAK_WIDTH),
+                                 bounds_error=False)
+            detect_lines(self.lines, self.lambdas, self.data, self.err, fwhm_func=fwhm_func,
+                         calibration_lines_only=True)
```

Comment:
This whole block feels a bit surprising/scary to me. If there's nothing fitted yet, I think failing/returning `False` or something would be fine, or just plotting what you can too, without the calibrations applied, but actually detecting lines inside a plotting function is a bit weird, no?

Reply:
Yeah, but the fitted results are not stored in the .fits files. So when I load a Spectrum from a FITS I need to refit the lines. Should not be a problem for AuxTel.

Reply:
OK, fine by me 🙂